src/main/java/org/jitsi/protocol/xmpp/util/SSRCSignaling.java

```diff
@@ -343,6 +343,38 @@ public static boolean mergeVideoIntoAudio(MediaSourceMap peerSSRCs)
         return ownerMapping;
     }
 
+    /**
+     * From the given list of {@link SourceGroup}s select those that share
+     * the given MSID value.
+     *
+     * @param groups a {@link List} of {@link SourceGroup} to be filtered out.
+     * @param groupMsid a {@link String} with the MSID value to be used as
+     *        selector.
+     *
+     * @return a {@link List} of {@link SourceGroup}.
+     */
+    public static List<SourceGroup> selectWithMsid(
+        List<SourceGroup>    groups,
+        String               groupMsid)
+    {
+        if (StringUtils.isNullOrEmpty(groupMsid))
+        {
+            throw new IllegalArgumentException("Null or empty 'groupMsid'");
+        }
+
+        List<SourceGroup> result = new LinkedList<>();
```

Comment:
this could be:

```
return groups.stream()
       .filter(group -> group.getGroupMsid().equalsIgnoreCase(groupMsid)
       .collect(Collectors.toList());

```
which is a bit more concise and still readable i think, but i don't feel too strongly about changing it.

Reply:
I don't think we are at Java 8 already, but will add a FIXME, thanks !

Reply:
ah oops forgot about that. ok no worries.